tests/unit-ext/test-ext-arg.c

```diff
@@ -211,6 +215,44 @@ test_validator2_handler (const jerry_value_t func_obj_val __attribute__((unused)
   return jerry_create_undefined ();
 } /* test_validator2_handler */
 
+/**
+ * Convert all props of an JS object into native variables.
+ */
+static jerry_value_t
+test_validator_prop_handler (const jerry_value_t func_obj_val __attribute__((unused)), /**< function object */
+                             const jerry_value_t this_val __attribute__((unused)), /**< this value */
+                             const jerry_value_t args_p[], /**< arguments list */
+                             const jerry_length_t args_cnt __attribute__((unused))) /**< arguments length */
+{
+  bool native1 = false;
+  double native2 = 0;
+  double native3 = 3;
+
+  const char *name_p[] = {"prop1", "prop2", "prop3"};
+
+  jerryx_arg_t mapping[] =
+  {
+    jerryx_arg_boolean (&native1, JERRYX_ARG_COERCE, JERRYX_ARG_REQUIRED),
+    jerryx_arg_number (&native2, JERRYX_ARG_COERCE, JERRYX_ARG_REQUIRED),
+    jerryx_arg_number (&native3, JERRYX_ARG_COERCE, JERRYX_ARG_OPTIONAL)
+  };
+
+  jerry_value_t is_ok = jerryx_arg_transform_object_properties (args_p[0],
```

Comment:
let me try and pick your mind: what else do those custom validator do? do they always just call `jerryx_arg_transform_object_properties`? couldn't we have this function as
```c
jerry_value_t
jerryx_arg_transform_object_properties (jerryx_arg_js_iterator_t *js_arg_iter_p, 
                                        const jerryx_arg_t *c_arg_p) 
```
and have`name_p`, `name_cnt`, `c_arg_p`, `c_arg_cnt` in `c_arg_p->extra_info`?